go/protocol/chat1/extras.go

```diff
@@ -3153,6 +3183,30 @@ func (m AssetMetadata) IsType(typ AssetMetadataType) bool {
 	return mtyp == typ
 }
 
+func (m AssetMetadataImage) MarshalJSON() ([]byte, error) {
+	amps := make([]float64, 0, len(m.AudioAmps))
+	for _, amp := range m.AudioAmps {
+		if math.IsNaN(amp) {
+			amps = append(amps, 0)
+		} else {
+			amps = append(amps, amp)
+		}
+	}
+	width, err := json.Marshal(m.Width)
+	if err != nil {
+		return nil, err
+	}
+	height, err := json.Marshal(m.Height)
+	if err != nil {
+		return nil, err
+	}
+	mamps, err := json.Marshal(amps)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(fmt.Sprintf(`{"width":%s,"height":%s,"audioAmps":%s}`, width, height, mamps)), nil
```

Comment:
why not have a single struct here too

Reply:
Yeah I thought about it but felt bad about replicating the struct, but I guess that is basically done anyway.